api/v1alpha1/kogitoserverlessworkflow_types.go

```diff
@@ -308,20 +309,31 @@ type State struct {
 	Metadata            *[]Metadata      `json:"metadata,omitempty"`
 }
 
-// KogitoServerlessWorkflowSpec defines the desired state of KogitoServerlessWorkflow
+// KogitoServerlessWorkflowSpec defines the desired state of KogitoServerlessWorkflowApp
 type KogitoServerlessWorkflowSpec struct {
-	Constants   []Constant   `json:"conditions,omitempty"`
-	Secrets     *[]v1.Secret `json:"secrets,omitempty"`
-	Start       string       `json:"start"`
-	Timeouts    []Timeout    `json:"timeouts,omitempty"`
-	Errors      []Error      `json:"errors,omitempty"`
-	KeepActive  bool         `json:"keepActive,omitempty"`
-	Auth        Auth         `json:"auth,omitempty"`
-	Events      *[]Event     `json:"events,omitempty"`
-	Functions   []Function   `json:"functions,omitempty"`
-	AutoRetries bool         `json:"autoRetries,omitempty"`
-	Retries     Retry        `json:"retries,omitempty"`
-	States      []State      `json:"states"`
+	Constants    []Constant   `json:"conditions,omitempty"`
+	Secrets      *[]v1.Secret `json:"secrets,omitempty"`
+	Start        string       `json:"start"`
+	Timeouts     []Timeout    `json:"timeouts,omitempty"`
+	Errors       []Error      `json:"errors,omitempty"`
+	KeepActive   bool         `json:"keepActive,omitempty"`
+	Auth         Auth         `json:"auth,omitempty"`
+	Events       *[]Event     `json:"events,omitempty"`
+	Functions    []Function   `json:"functions,omitempty"`
+	AutoRetries  bool         `json:"autoRetries,omitempty"`
+	Retries      Retry        `json:"retries,omitempty"`
+	States       []State      `json:"states"`
+	CommonConfig CommonConfig `json:"commonConfig,omitempty"`
+}
+
+// CommonConfig variables used in the templates
+type CommonConfig struct {
```

Comment:
Let's add here a marker like a TODO to remove this from the Workflow spec since this is not part of the Workflow but are information needed for the deployment.

Otherwise if we would like to leave this details in this CRD it is probably better to use the Metadata for storing these details

Reply:
ok